tests/units/iop/test_iop_parser.py

```diff
@@ -43,19 +58,30 @@ def test_iop_record_validation(iop_parser, parsed_article_without_page_nr):
     iop_validate_record(enriched)
 
 
+@fixture
+def article_without_doi(shared_datadir):
+    with open(shared_datadir / "without_doi.xml") as f:
+        yield ET.fromstring(f.read())
+
+
+def parsed_article_without_doi(iop_parser, article_without_doi):
+    with raises(RequiredFieldNotFoundExtractionError):
+        return iop_parser._publisher_specific_parsing(article_without_doi)
+
+
 @fixture
 def articles(shared_datadir):
     files = ["example1.xml", "example2.xml"]
     articles = []
     for file in files:
         with open(shared_datadir / file) as f:
             articles.append(ET.fromstring(f.read()))
-    return articles
+    yield articles
 
 
 @fixture
 def parsed_articles(iop_parser, articles):
-    return [iop_parser._publisher_specific_parsing(article) for article in articles]
+    yield [iop_parser._publisher_specific_parsing(article) for article in articles]
```

Comment:
```suggestion
     for article in articles:
         yield iop_parser._publisher_specific_parsing(article)
```

Reply:
why it is better?

Reply:
https://stackoverflow.com/questions/59536684/pytest-fixture-yield-list-of-dict-throws-yield-fixture-function-has-more-than-o
...But fixture functions can only yield exactly one value....